pkg/vsock/handshake_windows.go

```diff
@@ -0,0 +1,123 @@
+package vsock
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/linuxkit/virtsock/pkg/hvsock"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows/registry"
+)
+
+// GetVMGUID retrieves the GUID for a correct hyper-v VM (most likely WSL).
+// It performs a handshake with a running vsock-peer in the WSL distro
+// to make sure we establish the AF_VSOCK connection with a right VM.
+func GetVMGUID(seed string, handshakePort uint32, timeout <-chan time.Time) (hvsock.GUID, error) {
```

Comment:
Future improvement (not in this PR): on every iteration, re-check the list of VM GUIDs (so we can capture VMs that were not running at the start of the loop).